Feat/conCalendar 캘린더 연동 기능

yellTa opened the pull request:
캘린더 연동 기능 추가

Commit:
first commit feat/conCalendar

Commit:
테스트를 위한 SpringSecurity옵션 모두 끄기

Commit:
feat: HourplaceAccount 계정 저장하기

hourpalce의 계정을 저장한다.

CalendarController.java
POST로 들어온 값을 비지니스 로직을 이용해 저장한다.

CalendarControllerTest.java
Spring test문서 POST로 값을 보내고
저장하는 코드

CalendarServiceImpl.java
hourplaceAccount 저장 로직 구현
사용자가 없으면 새로 저장하고, 있으면 들어온 값을 update한다.

HourplaceAccount.java
Account Table 객체 생성

HourplaceAccountRepository.java
SpacecloudAccountRepository
리포지토리

Commit:
feat: spacecloud 계정 정보 로직
CalendarControllerTest.java
띄어쓰기 삭제

CalendarServiceImpl
Spacecloud 계정 정보 저장 로직 수행
계정이 있으면 update

SpacecloudAccount.java
spaceCloud Entity생성

SpacecloudAccountRepository
findByuserId메서드를 통해 userID 가져옴

Commit:
feat :hourplace로그인 인증 추가
CalendarController.java
userId를 받아오는 부분을 제거
AccessToken에서 userId 뽑아서 사용하는 것으로 수행

CalendarServiceImpl.java
hourplace로그인 검증 로직 추가 ResponseEntity로 결과 전달
성공 -> 응답코드 200
실해 -> 응다봌드 401 메세지 "계졍이 일치하지 않습니다."

CrawlingLogic
hourplace혹은 spacecloud의 플랫폼 로그인 결과를 가져오고 판별함

실패시 null값의 driver반환 성공시 로그인이 된 상태의 driver반환

CrawlingLogicTest.java
테스트 코드 추가

HourplaceCrawling.java
아워플레이스 로그인 크롤링 로직 추가

Commit:
feat: spacecloud 로그인 검증 로직 구현

CalendarServiceImpl.java
spacecloud의 정 저장 이때 객체의 매개변수를 잘못입력했던 부분 수정

SpacecloudAccount.java
spacecloud 매개변수 이름 변경

CrawlingLogic
spacecloud 로그인 로직 수행
Driver가 null값이면 로그인 실패

SpacecloudCrawling
크롤링 기능 수행, 로그인이 완료되면 예약 가져오는 페이지까지 이동 실패시 드라이버 종료 후 null값인 driver return

SpacecloudAccountRepository
userId를 통해 객체를 반환하는 로직 수행

CrawlingLogicTest.java
loginCheck3로 spacecloud 플랫폼 정보, userId로 정보 가져오기 수행

Commit:
feat: 연동시 서버 오류가 생겼을 떄 Exception추가

HourplaceCrawling
LoginFailedException.java
SpacecloudAccountRepository
SpacecloudCrawling

CalendarServiceImpl.java
CrawlingLogic
HourplaceCrawling
SpacecloudCrawling.java
Exception추가 try catch를 통해 Excepton을 받고 처리
Exception은 크롤링 도중 에러가 발생한 경우

LoginFailedException.java
Custom Exception추가

chore
SpacecloudCrawling
띄어쓰기 제거

Commit:
feat : 크롤링 로직 수행하기

CalendarServiceImpl.java
spacecloud 부분 알 수 없는 계정일 떄 try catch문으로 LoginFailedException을 받고 예외 처리 response 생성후 return
서버에서 확인 위해 로그 추가

CrawlingLogic.java
크롤링 로직 추가 hourplace, spacecloud추가

CrawlingLogicTest.java
크롤링 로직 테스트  추가
userID를 임의로 변경해서 수행
JWT토큰에서 userId를 가져오기 때문

HourplaceCrawling.java
hourplace 크롤링 로직

SpacecloudCrawling.java
spacecloud크롤링 로직

Commit:
feat: 크롤링시 예약 번호, 링크, 사용자 정보 가져오기 추가 Crawling 수행시 예외 처리 추가

CalendarServiceImpl.java
크롤링 수행시 에러 발생 처리 추가

CrawlingException.java
크롤링 예외 처리 Exception

CrawlingLogicTest.java
크롤링 로직 테스트를 위해 작성
수행시 userId를 토큰에서 뽑아서 사용하지 않고 임의로 CalendarServiceImpl.java에서 userId를 지정해서 사용

HourplaceCrawling.java
SpacecloudCrawling.java
예외상황 로그추가 및 CrawlingExceptino처리
에약자명, 링크 추가

<style>
CrawlingLogic
띄어쓰기 제거

Bumnote commented on src/main/java/com/zerospace/zerospace/Const/Const.java:
```diff
@@ -0,0 +1,10 @@
+    public static final long ACCESS_TOKEN_VALIDITY_SECONDS = 20 * 60;
+    public static final long REFRESH_TOKEN_VALIDITY_SECONDS = 30 * 24 * 60 * 60;
+
+    public static final String SECRET_KEY = "zerosapceOur256SpecialKeyShouldBeVerySecure";
```
개인정보는 숨겨보죠! application.properties 또는 application.yml 파일에 넣고, @Value 어노테이션을 사용하시면 좋을 것 같습니다!

Commit:
chore : application.properties에 상수 설정하기
상수를 설정하고 properties파일을 숨김

Commit:
ignore application.properties file

Bumnote commented on src/main/java/com/zerospace/zerospace/Const/Const.java:
```diff
@@ -0,0 +1,21 @@
+    public static final long ACCESS_TOKEN_VALIDITY_SECONDS = 20 * 60;
+    public static final long REFRESH_TOKEN_VALIDITY_SECONDS = 30 * 24 * 60 * 60;
+
+    @Value("${jwt.secret-key}")
+    private String secretKeyValue;
+
+    @Value("${jwt.access-token-name}")
+    private String accessTokenNameValue;
+
+    @Value("${jwt.refresh-token-name}")
+    private String refreshTokenNameValue;
```
오호 훨씬 보기 좋아졌습니다!

yellTa replied:
감사합니닷 ㅎㅎㅎ

yellTa replied:
저희 FILTER 때문에 다시 뺴서 써야됩니다 ㅋ... Application.properties에 상수 선언 못합니다. Spring애플리케이션으로 진입하기 전에 상수를 사용해서요

Bumnote replied:
application.properties 상수 선언 관련해서 이야기 나눠보면 좋을 것 같네요 ㅎㅎㅎ

yellTa replied:
Filter랑 Spring에서 Filter랑 똑같은 행동을 하는 친구를 알게 된다면 왜 저래 썻느지... 그리고 저걸 어떻게 고쳐야할지 알게 되실겁니다 ㅎㅎ

Bumnote commented on src/main/java/com/zerospace/zerospace/domain/Member.java:
```diff
@@ -0,0 +1,26 @@
+
+    private String nickName;
+
+    public Member(){
+    }
```
이 부분은 @NoArgsConstructor 어노테이션을 활용해보는 게 어떠실까요 ??

Bumnote commented on src/main/java/com/zerospace/zerospace/domain/Member.java:
```diff
@@ -0,0 +1,26 @@
+
+    public Member(){
+    }
+    public Member(String userId, String email, String nickName){
+        this.userId = userId;
+        this.email = email;
+        this.nickName = nickName;
+
+    }
```
이 부분도 @AllArgsConstructor를 사용하면 코드가 깔끔해질 것 같은데, 이렇게 다 적어두신 이유가 있을까요 ?

yellTa replied:
일단 생성자를 생성해야하는 것은 알고있었는데 위의 두 애노테이션은 몰랐습니다
바로 수정하겠습니다 알려주셔서 감사합니다 ㅎㅎ

Bumnote reviewed (commented):
간단한 부분에 대해서만 이야기 해보았습니다 :)  👍

Commit:
style: DTO Annotation을 이용한 코드 리팩토링

@NoArgsConstructor
@AllArgsConstructor
두 애노테이션을 이용해 DTO에 기본 생성자 + 파라미터를 받는 생성자 생성

HourplaceAccount
Member
SpacecloudAccount에 각각 적용 됨

Commit:
feat: hourplace, spacecloud 크롤링로직 추가

feat : HourplaceCrawling.java
날짜+ 시간으로 파싱하는 로직 추가

feat : SpacecloudCrawling.java
날짜+ 시간 으로 파싱하는 데이터 추가

style : CrawlingLogic.java
hourplace로그를 spacecloud로 잘못 표기해 수정

Commit:
feat: headless 크롤링 로직 추가
(API 확인 해봐야함)
result를 list를 받아서 처리
hourplace선 처리 후에 spacecloud 후처리

Bumnote commented on src/main/java/com/zerospace/zerospace/SecurityFilter.java:
```diff
@@ -0,0 +1,46 @@
+public class SecurityFilter {
+
+
+    private final JwtAuthFilter jwtAuthFilter;
+    private final CORSFilter corsFilter;
+
+    public SecurityFilter(JwtAuthFilter jwtAuthFilter, CORSFilter corsFilter) {
+        this.jwtAuthFilter = jwtAuthFilter;
+        this.corsFilter = corsFilter;
+    }
```
생성자 주입 방법으로 하셨군요! 그런데 이 부분도 
@RequiredArgsConstructor 어노테이션을 활용하시면 생성자를 따로 작성 안해도 처리가 가능합니다!

저도 이 부분 관련하여 멘토님한테 어노테이션을 추천받아서 한번 알아보세요! :) 
(+ 밑에 JwtFillter class 관련해서는 RequiredArgsConstructor 어노테이션을 사용하셨는데 여기서는 사용 안하신 이유가 따로 있는지요 ??)

yellTa replied:
@COnfiguration는 해당 애노테이션을 사용할 수 없습니다. 빈 주입이 제대로 되지 않아서 @Autowired를 사용해 주입을 합니당

Bumnote commented on src/main/java/com/zerospace/zerospace/domain/HourplaceAccount.java:
```diff
@@ -0,0 +1,20 @@
+@NoArgsConstructor
+@AllArgsConstructor
+public class HourplaceAccount {
+    @Id
+    private String userId;
```
여기 PK는 auto_increment 형식이 아닌 건가요 ??
GenertatedValue() 가 없는 이유가 따로 있을까요 ?

yellTa replied:
저희 PK 랜덤 스트링입니다 auto_increment하면 아이디 유추 가능성 있어서 빼고 index용으로만 써요

Bumnote replied:
아하 그렇군요! PK 랜덤 스트링에 대해서 저도 한번 공부해보겠습니다! :)

Commit:
feat: zzerospace redirect 링크 SecurityFilter 지정
redirect 주소를 8080포트에 oauth2/authorizeation/kakao 로 변경

JWTTokenService.java
request.getHeader로  JWT토큰에서 userId 가져오기

나머지는 모두 log나 띄어쓰기 제거임

Commit:
feat: 크롤링 정보 저장
CalendarServiceImpl.java
크롤링한 정보 저장하는 로직 추가

CalendarInfoRepository.java
ReservationNubmer기준으로 Calendar객체를 가져옴

CalendarInfo
HourplaceCrawling.java
SpacecloudCrawling.java
reservation 변수 이름 오타 수정

Const
토큰 시간 몇 분인지 체크

Commit:
style : 띄어쓰기 제거

Commit:
feat : 카카오 OAuth 리다이렉트 링크 추가

LoginController.java
OAuth 리다이렉트 링크 추가

OAuthSuccessHandler.java
Httponly쿠키 생성, accesstoken생성 후 카카오 redirect URI로 forward

Const.java
application.properties에서 가져온 값으로 사용설정

CORSFilter.java
origin에 front링크 추가 origin에서 들어오는 것들은 CORS검사 제외

JwtAuthFilter.java
jwt필터 로그 설정

Commit:
feat : OAuth 관련 셋업 설정

Const.java
상수 설정

JwtAuthFilter.java
accessToken, refreshToken 생성

LoginController
Jwt 생성

OAuthSuccessHandler.java
forward로그 추가 하지만 사용안하게 될듯

SecurityFilter.java
loginResult 리다이렉트 보내기

Commit:
api연동으로 바뀐 것들

Commit:
api연동때 바뀐 사항들

Commit:
JWT토큰 복구

Commit:
CORS설정 완료

Commit:
에러 잡기 완료

Commit:
feat: 크롤링 로직 추가

Commit:
feat: 크롤링로직 추가

Commit:
JWT토큰에서 userId 가져오는거로 변경
띄어쓰기 제거
